From 22b1e0897af977cc96af926c730ff948bd120bb5 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Fri, 31 Mar 2023 09:39:49 +0200
Subject: [PATCH] Fix build for GCC 13 + make GitHub Actions cover build with
 GCC 13 (#586)

* include missing <cstdint>

gcc 13 moved some includes around and as a result <cstdint> is no
longer transitively included [1]. Explicitly include it for
uint8_t.

[1] https://gcc.gnu.org/gcc-13/porting_to.html#header-dep-changes

Signed-off-by: Khem Raj <raj.khem@gmail.com>

* docker.yml: Cover openSUSE Tumbleweed with GCC 13+

Based on scripts/docker/build_on_centos_8_2.Dockerfile .

---------

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Co-authored-by: Khem Raj <raj.khem@gmail.com>
---
 .github/workflows/docker.yml                  |  1 +
 .../build_on_opensuse_tumbleweed.Dockerfile   | 50 +++++++++++++++++++
 src/Library/Base64.cpp                        |  1 -
 src/Library/Base64.hpp                        |  1 +
 4 files changed, 52 insertions(+), 1 deletion(-)
 create mode 100644 scripts/docker/build_on_opensuse_tumbleweed.Dockerfile

diff --git a/src/Library/Base64.cpp b/src/Library/Base64.cpp
index ddb28dce..0246a134 100644
--- a/src/Library/Base64.cpp
+++ b/src/Library/Base64.cpp
@@ -22,7 +22,6 @@
 
 #include "Base64.hpp"
 #include <stdexcept>
-#include <cstdint>
 
 namespace usbguard
 {
diff --git a/src/Library/Base64.hpp b/src/Library/Base64.hpp
index 0947f214..e0c745cb 100644
--- a/src/Library/Base64.hpp
+++ b/src/Library/Base64.hpp
@@ -23,6 +23,7 @@
 #endif
 
 #include <string>
+#include <cstdint>
 #include <cstddef>
 
 namespace usbguard
